pkg/pkgrepository/app_deploy.go

```diff
@@ -30,13 +30,105 @@ func (a *App) deploy(tplOutput string) exec.CmdRunResult {
 			return exec.NewCmdRunResultWithErr(fmt.Errorf("Preparing kapp: %s", err))
 		}
 
-		return kapp.Deploy(tplOutput, a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
+		// TODO: should we inject this somewhere else? - there's some other rules that might interact with it...
+		fmt.Println("sending to kapp deploy: \n", appendRebaseRule(tplOutput))
+		return kapp.Deploy(appendRebaseRule(tplOutput), a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
 
 	default:
 		return exec.NewCmdRunResultWithErr(fmt.Errorf("Unsupported way to deploy"))
 	}
 }
 
+func appendRebaseRule(tplOutput string) string {
+	return tplOutput + `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+- ytt:
+    overlayContractV1:
+      overlay.yml: |
+        #@ load("@ytt:data", "data")
+        #@ load("@ytt:json", "json")
+        #@ load("@ytt:overlay", "overlay")
+        #@ load("@ytt:struct", "struct")
+
+        #@ def get_rev(annotations):
+        #@  if hasattr(annotations, "packaging.carvel.dev/repo-rev"):
```

Comment:
What is the `rev` in `packaging.carvel.dev/repo-rev` stand for?

Reply:
@neil-hickey good question,  especially as this is a place where the rebase rule is ahead of the tests -- 

it's a "revision counter" 

there's a related TODO in the list of test cases i haven't written yet:
https://github.com/vmware-tanzu/carvel-kapp-controller/pull/657/files#diff-3c7cc881b74c8228f1e705cf8b9ea60b146d078f33a94009a19ac70ea4c75fe3R374

but both are definitely in a shorthand only sensible to someone who was working on the prototype. :)